### PR TITLE
updated playlist image feature

### PR DESCRIPTION
**Pull request by andrewzadrozny**
now can use image urls from discogs, lastfm, and itunes

### CODE REVIEW

File: BasicMusicPlayer/app/src/main/java/data/artwork/PlaylistImager.kt [hunk 1]
```diff
@@ -1,44 +1,220 @@
...
+    suspend fun fetchDiscogsArtistImage(playcut: PlaylistDetails): String? {
+        try {
+            val artist = playcut.playcut.artistName
+            if (artist.equals("v/a", ignoreCase = true) || artist.equals("various artists", ignoreCase = true)){
+                return null
+            }
+            val response = DiscogsArtFetcher.discogsService.getArtist(
+                artist,
+                "tYvsaskeJxOQbWoZSSkh",
+                "vZuPZFFDerXIPrBfSNnNyDhXjpIUiyXi"
```
Comment (jakebromberg) [inline BasicMusicPlayer/app/src/main/java/data/artwork/PlaylistImager.kt:199]: Extract these into named constants.

File: BasicMusicPlayer/app/src/main/java/data/artwork/PlaylistImager.kt [hunk 2]
```diff
@@ -1,44 +1,220 @@
...
+        }
+        return null
+    }
 
-                    val imageUrl = searchResult.results.firstOrNull()?.thumb
+    // fetches the artist image url from discogs
+    suspend fun fetchDiscogsArtistImage(playcut: PlaylistDetails): String? {
+        try {
+            val artist = playcut.playcut.artistName
+            if (artist.equals("v/a", ignoreCase = true) || artist.equals("various artists", ignoreCase = true)){
```
Comment (jakebromberg) [inline BasicMusicPlayer/app/src/main/java/data/artwork/PlaylistImager.kt:193]: We should be able to fetch an image even if it's v/a, no?

File: BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/PlayerActivity.kt [hunk 3]
```diff
@@ -92,24 +85,24 @@ class PlayerActivity : AppCompatActivity() {
...
+                            playlistDetailsList.addAll(0, (updatedSublistWithImages.subList(0, 6)))
                             recyclerView.adapter?.notifyItemRangeChanged(0, 6)
                         }
                         // adds new entry to the playlist
                         else if (newEntry) {
                             println("addition to the playlist")
-                            val rangeToRemove = minOf(5, playlistDetailsList.size)
-                            playlistDetailsList.subList(0, rangeToRemove).clear()
-                            playlistDetailsList.addAll(0, (updatedFullList.subList(0, 6)))
+                            //val rangeToRemove = minOf(5, playlistDetailsList.size)
```
Comment (jakebromberg) [inline BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/PlayerActivity.kt:103]: Remove this?

File: BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/PlayerActivity.kt [hunk 4]
```diff
@@ -92,24 +85,24 @@ class PlayerActivity : AppCompatActivity() {
...
+                            playlistDetailsList.subList(0, 5).clear()
+                            playlistDetailsList.addAll(0, updatedSublistWithImages.subList(0, 6))
```
Comment (jakebromberg) [inline BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/PlayerActivity.kt:105]: What are these constants for? Could you extract them into variables and name them?

File: BasicMusicPlayer/app/src/main/java/data/artwork/PlaylistImager.kt (hunk 2 above)
Comment (andrewzadrozny) [reply BasicMusicPlayer/app/src/main/java/data/artwork/PlaylistImager.kt:193]: yes! my mistake

**Pull request closed without merging**